### ISSUE TITLE
ModelNode crashes when loading a model

### ISSUE THREAD

**Issue by Michael-Kr** [SceneView/sceneview-android#374]
Using the sample AR Model Viewer that depends on `io.github.sceneview:arsceneview:2.0.1` I tried to load and present a different model but the app crashed in the constructor of the `ModelNode`.  
Specifically in this code:
```
nodes.values.forEach { node ->
    node.parent = when (val parentEntity = node.parentEntity) {
        this.entity -> this
        else -> nodes.values.first { it.entity == parentEntity } // Crash happens here because first() can not find a value
    }
}
```
Here is the model I failed to use: https://sketchfab.com/3d-models/sphere-b31b12ffa93a40f48c9d991b6f168f4d
I am new to AR world so can not explain the issue well, but I assume that it is related to the internal structure of the .glb model I used (I used a few and they are quite standard). 

I managed to fix the issue by replacing the code with the following - 
```
nodes.values.forEach { node ->
    node.parent = when (val parentEntity = node.parentEntity) {
        this.entity -> this
        else -> nodes.values.firstOrNull { it.entity == parentEntity } ?: this
    }
}
```
In other words, I assume that if we can not find a parent - `this` is the parent. 

Is this fix valid or something else has to be done?

**Comment by dhaval-android** [comment]
I am also getting crashes when I try to load the model with the animation using the bellow code and you find the URL of the model which I am trying to implement plz help if possible.

```

bindingView.sceneView.addChildNode(
            AnchorNode(bindingView.sceneView.engine, anchor)
                .apply {
                    isEditable = false
                    lifecycleScope.launch {
                        isLoading = true
                        bindingView.sceneView.modelLoader.loadModelInstance(
                            "models/stature_anim.glb"
                        )?.let { modelInstance ->
                            addChildNode(
                                ModelNode(
                                    autoAnimate = true,
                                    modelInstance = modelInstance,
//                                    centerOrigin = Position(y = -0.5f)
                                    // Scale to fit in a 0.5 meters cube
//                                    scaleToUnits = 1.5f,
                                    // Bottom origin instead of center so the model base is on floor
//                                    centerOrigin = Position(y = -0.5f)
                                ).apply {
                                    isEditable = false
                                }
                            )
                        }
                        isLoading = false
                    }
                    anchorNode = this
                }
        )
```

Model trying to load
https://drive.google.com/file/d/1hEFmAUZysTQ6xE0L8YgowY1fKwxSWQmy/view?usp=sharing


I am using the latest version 
2.0.1


```
Crash Log

java.util.NoSuchElementException: Collection contains no element matching the predicate.
                                                             	at io.github.sceneview.node.ModelNode.<init>(ModelNode.kt:416)
                                                             	at io.github.sceneview.node.ModelNode.<init>(ModelNode.kt:36)
                                                             	at com.vizitar.aroutdoor.MainActivity$addAnchorNode$1$1.invokeSuspend(MainActivity.kt:317)
                                                             	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                             	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                                                             	at android.os.Handler.handleCallback(Handler.java:942)
                                                             	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                             	at android.os.Looper.loopOnce(Looper.java:240)
                                                             	at android.os.Looper.loop(Looper.java:351)
                                                             	at android.app.ActivityThread.main(ActivityThread.java:8422)
                                                             	at java.lang.reflect.Method.invoke(Native Method)
                                                             	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
                                                             	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
                                                             	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@bba82a9, Dispatchers.Main.immediate]
```

**Comment by Michael-Kr** [comment]
@dhaval-android this is exactly the crash I was getting. Currently, I use the fix I made in a forked project. Probably it should be applied to the library but I am not sure. Maybe @ThomasGorisse can clarify.

**Comment by ThomasGorisse** [comment]
Nice catch @Michael-Kr !!!

I have pushed the possible temporary fix here: https://github.com/SceneView/sceneview-android/commit/aa8eda5408a7dae2b99d019f01f1cf08b129a8b4

I'm waiting for a Filament answer here to know if the orphan glTF entity is an issue or a possible world parenting behavior: https://github.com/google/filament/discussions/7445

**Comment by dhaval-android** [comment]
@Michael-Kr  @ThomasGorisse Now crash is not happen But if we are making `autoAnimate = true` or  `playAnimation(0)` But then also animation is not working ( I checked that we are getting animation count morethen 0 ) 
Bellow is the model which I am trying to use
https://drive.google.com/file/d/1hEFmAUZysTQ6xE0L8YgowY1fKwxSWQmy/view?usp=sharing

**Comment by dhaval-android** [comment]
Finally, it's working!!
But For some of the models `autoAnimate = true` is not working so we need to play the animation manually, it happened in my case also for the above module so played all the animation manually.

```

for (pos in 0..<animator.animationCount){
   playAnimation(pos, true)
 }
```
